packages/gatsby/src/query/query-compiler.js

```diff
@@ -396,15 +382,44 @@ const processDefinitions = ({
       continue
     }
 
+    const printedDocument = print(document)
+    // Check for duplicate page/static queries in the same component.
+    // (config query is not a duplicate of page/static query in the component)
+    // TODO: make sure there is at most one query type per component (e.g. one config + one page)
```

Comment:
This TODO still a TODO?

Reply:
Yup, we didn't touch this part in this PR

Reply:
This comment also is not exactly relevant to what this PR is doing, so it should stay